server.go

```diff
@@ -1316,6 +1320,32 @@ func (s *Server) Authenticate(username, password string) (*User, error) {
 	return u, nil
 }
 
+// Authenticate by secret file
+func (s *Server) AuthenticateBySecret(secret string) (*User, error) {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	// load content of the file (first line only)
+	file, err := os.Open(s.SecretFile)
```

Comment:
We should load this file when the server starts up.  I don't think we want to read from disk during an handler request.